// @pkt2022/uikit/dist/index.esm.js
import styled, { keyframes, css, createGlobalStyle } from 'styled-components';
import { background, border, layout, position, space, flexbox, variant, typography } from 'styled-system';
import React, { Children, cloneElement, isValidElement, useMemo, useReducer, useCallback, useEffect, useState, useRef, createContext, useContext } from 'react';
import noop from 'lodash/noop';
import debounce from 'lodash/debounce';

/*! *****************************************************************************
Copyright (c) Microsoft Corporation.

Permission to use, copy, modify, and/or distribute this software for any
purpose with or without fee is hereby granted.

THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
PERFORMANCE OF THIS SOFTWARE.
***************************************************************************** */

var __assign = function() {
    __assign = Object.assign || function __assign(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};

function __rest(s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
}

/** @deprecated */
function __spreadArrays() {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
}

function __makeTemplateObject(cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
}

var Box = styled.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n"], ["\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n"])), background, border, layout, position, space);
var templateObject_1;

var Flex = styled(Box)(templateObject_1$1 || (templateObject_1$1 = __makeTemplateObject(["\n  display: flex;\n  align-items: center;\n  ", "\n"], ["\n  display: flex;\n  align-items: center;\n  ", "\n"])), flexbox);
var templateObject_1$1;

// const getBackgroundColor = ({ theme, variant }: StyledButtonMenuProps) => {
//   return theme.colors[variant === variants.SUBTLE ? "input" : "tertiary"];
// };
var StyledButtonMenu = styled.div(templateObject_1$2 || (templateObject_1$2 = __makeTemplateObject(["\n  background-color: transparent;\n  border-radius: 16px;\n  display: inline-flex;\n\n  & > button + button,\n  & > a + a {\n    margin-left: 2px; // To avoid focus shadow overlap\n  }\n"], ["\n  background-color: transparent;\n  border-radius: 16px;\n  display: inline-flex;\n\n  & > button + button,\n  & > a + a {\n    margin-left: 2px; // To avoid focus shadow overlap\n  }\n"])));
var templateObject_1$2;

var scales = {
    MD: "md",
    SM: "sm",
    XS: "xs",
};
var variants = {
    PRIMARY: "primary",
    SECONDARY: "secondary",
    TERTIARY: "tertiary",
    TEXT: "text",
    DANGER: "danger",
    SUBTLE: "subtle",
    SUCCESS: "success",
    LIGHT: "light",
};

var ButtonMenu = function (_a) {
    var _b = _a.activeIndex, activeIndex = _b === void 0 ? 0 : _b, _c = _a.scale, scale = _c === void 0 ? scales.MD : _c, _d = _a.variant, variant = _d === void 0 ? variants.PRIMARY : _d, onItemClick = _a.onItemClick, children = _a.children;
    return (React.createElement(StyledButtonMenu, { variant: variant }, Children.map(children, function (child, index) {
        return cloneElement(child, {
            isActive: activeIndex === index,
            onClick: onItemClick ? function () { return onItemClick(index); } : undefined,
            scale: scale,
            variant: variant,
        });
    })));
};

var getExternalLinkProps = function () { return ({
    target: "_blank",
    rel: "noreferrer noopener",
}); };

var _a, _b;
var scaleVariants = (_a = {},
    _a[scales.MD] = {
        height: "48px",
        padding: "0 24px",
    },
    _a[scales.SM] = {
        height: "32px",
        padding: "0 16px",
    },
    _a[scales.XS] = {
        height: "20px",
        fontSize: "14px",
        padding: "0 8px",
    },
    _a);
var styleVariants = (_b = {},
    _b[variants.PRIMARY] = {
        background: "#4BAD29",
        borderRadius: "10px",
        //  bg: primary
        color: "#FFFFFF",
    },
    _b[variants.SECONDARY] = {
        backgroundColor: "#219653",
        borderRadius: "10px",
        color: "#FFFFFF",
        boxShadow: "none",
    },
    _b[variants.TERTIARY] = {
        backgroundColor: "transparent",
        borderRadius: "8px",
        color: "#333333",
        boxShadow: "none",
        border: "none",
    },
    _b[variants.SUBTLE] = {
        backgroundColor: "textSubtle",
        color: "backgroundAlt",
    },
    _b[variants.DANGER] = {
        background: "linear-gradient(95.95deg, #085A7E -4.12%, #2F80ED 44.69%, #56CCF2 97.73%) !important",
        color: "white",
    },
    _b[variants.SUCCESS] = {
        background: 'radial-gradient(63.11% 63.11% at 31.97% 19.67%, rgba(255, 255, 255, 0.6) 0%, rgba(255, 255, 255, 0) 69.79%, rgba(255, 255, 255, 0) 100%)',
        backgroundColor: "#083753",
        color: "rgba(255, 255, 255, 0.85)",
        textShadow: '0.433026px 0.433026px 0.433026px rgba(0, 0, 0, 0.25)',
    },
    _b[variants.TEXT] = {
        backgroundColor: "transparent",
        color: "primary",
        boxShadow: "none",
    },
    _b[variants.LIGHT] = {
        backgroundColor: "#082938;",
        borderRadius: "10px",
        color: "#fff",
    },
    _b);

var getDisabledStyles = function (_a) {
    var $isLoading = _a.$isLoading; _a.theme;
    if ($isLoading === true) {
        return "\n      &:disabled,\n      &.button--disabled {\n        cursor: not-allowed;\n        color: purple;\n      }\n    ";
    }
    return "\n    &:disabled,\n    &.button--disabled {\n      background-color: pink;\n      border-color: blue;\n      box-shadow: none;\n      color: yellow;\n      cursor: not-allowed;\n    }\n  ";
};
/**
 *
  

 */
/**
 * This is to get around an issue where if you use a Link component
 * React will throw a invalid DOM attribute error
 * @see https://github.com/styled-components/styled-components/issues/135
 */
var getOpacity = function (_a) {
    var _b = _a.$isLoading, $isLoading = _b === void 0 ? false : _b;
    return $isLoading ? ".5" : "1";
};
var StyledButton = styled.button(templateObject_1$3 || (templateObject_1$3 = __makeTemplateObject(["\n  align-items: center;\n  border: 0;\n  border-radius: 10px;\n  box-shadow: 0px -1px 0px 0px rgba(14, 14, 44, 0.4) inset;\n  cursor: pointer;\n  display: inline-flex;\n  font-family: inherit;\n  font-size: 16px;\n  font-weight: 600;\n  justify-content: center;\n  letter-spacing: 0.03em;\n  line-height: 1.3;\n  opacity: ", ";\n  outline: 0;\n  //\n  background: radial-gradient(\n    63.11% 63.11% at 31.97% 19.67%,\n    rgba(255, 255, 255, 0.6) 0%,\n    rgba(255, 255, 255, 0) 69.79%,\n    rgba(255, 255, 255, 0) 100%\n  );\n  background-color: #083753;\n  color: rgba(255, 255, 255, 0.85);\n  text-shadow: 0.433026px 0.433026px 0.433026px rgba(0, 0, 0, 0.25);\n  transition: all 0.2s, opacity 0.2s;\n\n  &:hover:not(:disabled):not(.button--disabled):not(.button--disabled):not(:active) {\n    opacity: 0.9;\n    /* background-color: #6fcf97; */\n    box-shadow: 0px 6px 7px rgba(0, 0, 0, 0.25);\n  }\n\n  &:active:not(:disabled):not(.button--disabled):not(.button--disabled) {\n    opacity: 0.9;\n    transform: translateY(1px);\n    box-shadow: none;\n    color: rgba(255, 255, 255, 1);\n  }\n\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n"], ["\n  align-items: center;\n  border: 0;\n  border-radius: 10px;\n  box-shadow: 0px -1px 0px 0px rgba(14, 14, 44, 0.4) inset;\n  cursor: pointer;\n  display: inline-flex;\n  font-family: inherit;\n  font-size: 16px;\n  font-weight: 600;\n  justify-content: center;\n  letter-spacing: 0.03em;\n  line-height: 1.3;\n  opacity: ", ";\n  outline: 0;\n  //\n  background: radial-gradient(\n    63.11% 63.11% at 31.97% 19.67%,\n    rgba(255, 255, 255, 0.6) 0%,\n    rgba(255, 255, 255, 0) 69.79%,\n    rgba(255, 255, 255, 0) 100%\n  );\n  background-color: #083753;\n  color: rgba(255, 255, 255, 0.85);\n  text-shadow: 0.433026px 0.433026px 0.433026px rgba(0, 0, 0, 0.25);\n  transition: all 0.2s, opacity 0.2s;\n\n  &:hover:not(:disabled):not(.button--disabled):not(.button--disabled):not(:active) {\n    opacity: 0.9;\n    /* background-color: #6fcf97; */\n    box-shadow: 0px 6px 7px rgba(0, 0, 0, 0.25);\n  }\n\n  &:active:not(:disabled):not(.button--disabled):not(.button--disabled) {\n    opacity: 0.9;\n    transform: translateY(1px);\n    box-shadow: none;\n    color: rgba(255, 255, 255, 1);\n  }\n\n  ", "\n  ",
    "\n  ",
    "\n  ", "\n  ", "\n"])), getOpacity, getDisabledStyles, variant({
    prop: "scale",
    variants: scaleVariants,
}), variant({
    variants: styleVariants,
}), layout, space);
var templateObject_1$3;

var Button = function (props) {
    var startIcon = props.startIcon, endIcon = props.endIcon, external = props.external, className = props.className, isLoading = props.isLoading, disabled = props.disabled, children = props.children; props.isTrans; var rest = __rest(props, ["startIcon", "endIcon", "external", "className", "isLoading", "disabled", "children", "isTrans"]);
    var internalProps = external ? getExternalLinkProps() : {};
    var isDisabled = isLoading || disabled;
    var classNames = className ? [className] : [];
    if (isLoading) {
        classNames.push("button--loading");
    }
    if (isDisabled && !isLoading) {
        classNames.push("button--disabled");
    }
    return (React.createElement(StyledButton, __assign({ "$isLoading": isLoading, className: classNames.join(" "), disabled: isDisabled }, internalProps, rest),
        React.createElement(React.Fragment, null,
            isValidElement(startIcon) &&
                cloneElement(startIcon, {
                    mr: "0.5rem",
                }),
            children,
            isValidElement(endIcon) &&
                cloneElement(endIcon, {
                    ml: "0.5rem",
                }))));
};
Button.defaultProps = {
    isLoading: false,
    external: false,
    variant: variants.PRIMARY,
    scale: scales.MD,
    disabled: false,
};

var InactiveButton = styled(Button)(templateObject_1$4 || (templateObject_1$4 = __makeTemplateObject(["\n  background-color: transparent;\n  color: #000;\n  &:hover:not(:disabled):not(:active) {\n    background-color: transparent;\n  }\n"], ["\n  background-color: transparent;\n  color: #000;\n  &:hover:not(:disabled):not(:active) {\n    background-color: transparent;\n  }\n"])));
var ButtonMenuItem = function (_a) {
    var _b = _a.isActive, isActive = _b === void 0 ? false : _b, _c = _a.variant, variant = _c === void 0 ? variants.PRIMARY : _c, as = _a.as, props = __rest(_a, ["isActive", "variant", "as"]);
    if (!isActive) {
        return (React.createElement(InactiveButton, __assign({ forwardedAs: as, variant: "tertiary", colorKey: variant === variants.PRIMARY ? "primary" : "textSubtle" }, props)));
    }
    return React.createElement(Button, __assign({ as: as, variant: variant }, props));
};
var templateObject_1$4;

var IconButton = styled(Button)(templateObject_1$5 || (templateObject_1$5 = __makeTemplateObject(["\n  padding: 0;\n  width: ", ";\n"], ["\n  padding: 0;\n  width: ", ";\n"])), function (_a) {
    var scale = _a.scale;
    return scale && scale;
});
var templateObject_1$5;

var getFontSize = function (_a) {
    var fontSize = _a.fontSize, small = _a.small;
    return small ? "10px" : fontSize || "14px";
};
var Text = styled.div(templateObject_1$6 || (templateObject_1$6 = __makeTemplateObject(["\n  color: ", ";\n  font-size: ", ";\n  font-weight: ", ";\n  letter-spacing: 0.04em;\n  line-height: 1.5;\n  font-family: Gilroy;\n  ", "\n\n  ", "\n  ", "\n  ", "\n"], ["\n  color: ", ";\n  font-size: ", ";\n  font-weight: ", ";\n  letter-spacing: 0.04em;\n  line-height: 1.5;\n  font-family: Gilroy;\n  ",
    "\n\n  ", "\n  ", "\n  ", "\n"])), function (_a) {
    var color = _a.color;
    return color && color;
}, getFontSize, function (_a) {
    var bold = _a.bold;
    return (bold ? 700 : 500);
}, function (_a) {
    var ellipsis = _a.ellipsis;
    return ellipsis &&
        "white-space: nowrap;\n    overflow: hidden;\n    text-overflow: ellipsis;";
}, space, typography, layout);
Text.defaultProps = {
    color: "#363636",
    small: false,
    ellipsis: false,
};
var templateObject_1$6;

var tags = {
    H1: "h1",
    H2: "h2",
    H3: "h3",
    H4: "h4",
    H5: "h5",
    H6: "h6",
};
var scales$1 = {
    SM: "sm",
    MD: "md",
    LG: "lg",
    XL: "xl",
    XXL: "xxl",
};

var _a$1;
var style = (_a$1 = {},
    _a$1[scales$1.SM] = {
        fontSize: "18x",
        fontSizeLg: "18px",
    },
    _a$1[scales$1.MD] = {
        fontSize: "20px",
        fontSizeLg: "20px",
    },
    _a$1[scales$1.LG] = {
        fontSize: "24px",
        fontSizeLg: "24px",
    },
    _a$1[scales$1.XL] = {
        fontSize: "28px",
        fontSizeLg: "28px",
    },
    _a$1[scales$1.XXL] = {
        fontSize: "40px",
        fontSizeLg: "40px",
    },
    _a$1);
var Heading = styled(Text).attrs({ bold: true })(templateObject_1$7 || (templateObject_1$7 = __makeTemplateObject(["\n  font-size: ", ";\n  font-weight: 700;\n  line-height: 1.5;\n  font-family: GilroyBold;\n"], ["\n  font-size: ", ";\n  font-weight: 700;\n  line-height: 1.5;\n  font-family: GilroyBold;\n"])), function (_a) {
    var scale = _a.scale;
    return style[scale || scales$1.MD].fontSize;
});
Heading.defaultProps = {
    as: tags.H1,
};
var templateObject_1$7;

var scales$2 = {
    SM: "sm",
    MD: "md",
    LG: "lg",
};

/**
 * Edric Phan code
 */
var getBoxShadow = function (_a) {
    var _b = _a.isSuccess, isSuccess = _b === void 0 ? false : _b, _c = _a.isWarning, isWarning = _c === void 0 ? false : _c; _a.theme;
    if (isWarning) {
        return "0px 0px 0px 1px #ED4B9E, 0px 0px 0px 4px rgba(237, 75, 158, 0.2)";
    }
    if (isSuccess) {
        return "0px 0px 0px 1px #31D0AA, 0px 0px 0px 4px rgba(49, 208, 170, 0.2)";
    }
    return "inset 0px -3px 2px -1px rgba(74, 74, 104, 0.1)";
};
var getHeight = function (_a) {
    var _b = _a.scale, scale = _b === void 0 ? scales$2.MD : _b;
    switch (scale) {
        case scales$2.SM:
            return "32px";
        case scales$2.LG:
            return "48px";
        case scales$2.MD:
        default:
            return "40px";
    }
};
var Input = styled.input(templateObject_1$8 || (templateObject_1$8 = __makeTemplateObject(["\n  background-color: transparent;\n  border: 0;\n  border-radius: 16px;\n  box-shadow: ", ";\n  color: #280d5f;\n  display: block;\n  font-size: 16px;\n  height: ", ";\n  outline: 0;\n  padding: 0 16px;\n  width: 100%;\n\n  &::placeholder {\n    color: #a1a3a6;\n    opacity: 0.6;\n  }\n\n  &:disabled {\n    background-color: #e9eaeb;\n    box-shadow: none;\n    color: #bdc2c4;\n    cursor: not-allowed;\n  }\n\n  &:focus:not(:disabled) {\n    box-shadow: none;\n  }\n"], ["\n  background-color: transparent;\n  border: 0;\n  border-radius: 16px;\n  box-shadow: ", ";\n  color: #280d5f;\n  display: block;\n  font-size: 16px;\n  height: ", ";\n  outline: 0;\n  padding: 0 16px;\n  width: 100%;\n\n  &::placeholder {\n    color: #a1a3a6;\n    opacity: 0.6;\n  }\n\n  &:disabled {\n    background-color: #e9eaeb;\n    box-shadow: none;\n    color: #bdc2c4;\n    cursor: not-allowed;\n  }\n\n  &:focus:not(:disabled) {\n    box-shadow: none;\n  }\n"])), getBoxShadow, getHeight);
Input.defaultProps = {
    scale: scales$2.MD,
    isSuccess: false,
    isWarning: false,
};
var templateObject_1$8;

var getPadding = function (scale, hasIcon) {
    if (!hasIcon) {
        return "16px";
    }
    switch (scale) {
        case scales$2.SM:
            return "32px";
        case scales$2.LG:
            return "56px";
        case scales$2.MD:
        default:
            return "48px";
    }
};
var StyledInputGroup = styled(Box)(templateObject_1$9 || (templateObject_1$9 = __makeTemplateObject(["\n  ", " {\n    padding-left: ", ";\n    padding-right: ", ";\n  }\n"], ["\n  ", " {\n    padding-left: ", ";\n    padding-right: ", ";\n  }\n"])), Input, function (_a) {
    var hasStartIcon = _a.hasStartIcon, scale = _a.scale;
    return getPadding(scale, hasStartIcon);
}, function (_a) {
    var hasEndIcon = _a.hasEndIcon, scale = _a.scale;
    return getPadding(scale, hasEndIcon);
});
var InputIcon = styled.div(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n  align-items: center;\n  display: flex;\n  height: 100%;\n  position: absolute;\n  top: 0;\n\n  ", "\n"], ["\n  align-items: center;\n  display: flex;\n  height: 100%;\n  position: absolute;\n  top: 0;\n\n  ",
    "\n"])), function (_a) {
    var isEndIcon = _a.isEndIcon, scale = _a.scale;
    return isEndIcon
        ? "\n    right: " + (scale === scales$2.SM ? "8px" : "16px") + ";\n  "
        : "\n    left: " + (scale === scales$2.SM ? "8px" : "16px") + ";\n  ";
});
var InputGroup = function (_a) {
    var _b = _a.scale, scale = _b === void 0 ? scales$2.MD : _b, startIcon = _a.startIcon, endIcon = _a.endIcon, children = _a.children, props = __rest(_a, ["scale", "startIcon", "endIcon", "children"]);
    return (React.createElement(StyledInputGroup, __assign({ scale: scale, width: "100%", position: "relative", hasStartIcon: !!startIcon, hasEndIcon: !!endIcon }, props),
        startIcon && React.createElement(InputIcon, { scale: scale }, startIcon),
        cloneElement(children, { scale: scale }),
        endIcon && (React.createElement(InputIcon, { scale: scale, isEndIcon: true }, endIcon))));
};
var templateObject_1$9, templateObject_2;

var rotate = keyframes(templateObject_1$a || (templateObject_1$a = __makeTemplateObject(["\n  from {\n    transform: rotate(0deg);\n  }\n  to {\n    transform: rotate(360deg);\n  }\n"], ["\n  from {\n    transform: rotate(0deg);\n  }\n  to {\n    transform: rotate(360deg);\n  }\n"])));
var spinStyle = css(templateObject_2$1 || (templateObject_2$1 = __makeTemplateObject(["\n  animation: ", " 2s linear infinite;\n"], ["\n  animation: ", " 2s linear infinite;\n"])), rotate);
var Svg = styled.svg(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n  align-self: center; // Safari fix\n  fill: ", ";\n  flex-shrink: 0;\n  ", "\n  ", "\n"], ["\n  align-self: center; // Safari fix\n  fill: ", ";\n  flex-shrink: 0;\n  ", "\n  ", "\n"])), function (_a) {
    var fill = _a.fill;
    return fill && fill;
}, function (_a) {
    var spin = _a.spin;
    return spin && spinStyle;
}, space);
Svg.defaultProps = {
    color: "text",
    width: "20px",
    xmlns: "http://www.w3.org/2000/svg",
    spin: false,
};
var templateObject_1$a, templateObject_2$1, templateObject_3;

var Icon = function (props) {
    return (React.createElement(Svg, __assign({ viewBox: "0 0 24 24" }, props),
        React.createElement("path", { d: "M8.11997 9.29006L12 13.1701L15.88 9.29006C16.27 8.90006 16.9 8.90006 17.29 9.29006C17.68 9.68006 17.68 10.3101 17.29 10.7001L12.7 15.2901C12.31 15.6801 11.68 15.6801 11.29 15.2901L6.69997 10.7001C6.30997 10.3101 6.30997 9.68006 6.69997 9.29006C7.08997 8.91006 7.72997 8.90006 8.11997 9.29006Z" })));
};

var Icon$1 = function (props) {
    return (React.createElement(Svg, __assign({ viewBox: "0 0 24 24" }, props),
        React.createElement("path", { d: "M8.11997 14.7101L12 10.8301L15.88 14.7101C16.27 15.1001 16.9 15.1001 17.29 14.7101C17.68 14.3201 17.68 13.6901 17.29 13.3001L12.7 8.7101C12.31 8.3201 11.68 8.3201 11.29 8.7101L6.69997 13.3001C6.30997 13.6901 6.30997 14.3201 6.69997 14.7101C7.08997 15.0901 7.72997 15.1001 8.11997 14.7101Z" })));
};

var Icon$2 = function (props) {
    return (React.createElement(Svg, __assign({ viewBox: "0 0 24 24" }, props, { fill: 'none' }),
        React.createElement("path", { d: "M10 3H3V10H10V3Z", stroke: "white", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round" }),
        React.createElement("path", { d: "M21 3H14V10H21V3Z", stroke: "white", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round" }),
        React.createElement("path", { d: "M21 14H14V21H21V14Z", stroke: "white", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round" }),
        React.createElement("path", { d: "M10 14H3V21H10V14Z", stroke: "white", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round" })));
};

var Icon$3 = function (props) {
    return (React.createElement(Svg, __assign({ viewBox: "0 0 25 24" }, props, { fill: "none" }),
        React.createElement("path", { d: "M16.5154 9.39996L7.51538 4.20996", stroke: "white", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round" }),
        React.createElement("path", { d: "M21.0154 16V8.00003C21.015 7.6493 20.9224 7.30483 20.7469 7.00119C20.5714 6.69754 20.3191 6.44539 20.0154 6.27002L13.0154 2.27002C12.7113 2.09449 12.3665 2.00208 12.0154 2.00208C11.6643 2.00208 11.3194 2.09449 11.0154 2.27002L4.01538 6.27002C3.71164 6.44539 3.45936 6.69754 3.28384 7.00119C3.10832 7.30483 3.01574 7.6493 3.01538 8.00003V16C3.01574 16.3508 3.10832 16.6952 3.28384 16.9989C3.45936 17.3025 3.71164 17.5547 4.01538 17.73L11.0154 21.73C11.3194 21.9056 11.6643 21.998 12.0154 21.998C12.3665 21.998 12.7113 21.9056 13.0154 21.73L20.0154 17.73C20.3191 17.5547 20.5714 17.3025 20.7469 16.9989C20.9224 16.6952 21.015 16.3508 21.0154 16Z", stroke: "white", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round" }),
        React.createElement("path", { d: "M3.2854 6.95996L12.0154 12.01L20.7454 6.95996", stroke: "white", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round" }),
        React.createElement("path", { d: "M12.0154 22.08V12", stroke: "white", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round" })));
};

var Icon$4 = function (props) {
    return (React.createElement(Svg, __assign({ viewBox: "0 0 26 24" }, props, { fill: "none" }),
        React.createElement("path", { fillRule: "evenodd", clipRule: "evenodd", d: "M12.1063 0.0717772C11.9946 0.125063 10.6688 1.28086 9.16015 2.64025L6.41716 5.11192L5.39392 5.11398C3.7503 5.11726 3.08081 5.22219 2.35346 5.59047C1.26423 6.14193 0.51697 7.06147 0.128989 8.32759C0.0191672 8.68605 0.0187608 8.70645 0.00336956 14.3396C-0.00673889 18.0609 0.00596017 20.1097 0.0406032 20.3365C0.310687 22.1048 1.58882 23.4934 3.32539 23.9052C3.70596 23.9955 4.16302 24 12.9445 24C21.7915 24 22.1761 23.9961 22.4629 23.9044C23.2904 23.6398 23.8613 23.0677 24.1852 22.1783C24.2794 21.9196 24.291 21.7448 24.3107 20.2898L24.3325 18.6864L24.4927 18.6322C24.7172 18.5563 25.2255 18.1596 25.4328 17.8987C25.6476 17.6282 25.8667 17.1518 25.9422 16.7907C26.0184 16.4263 26.0197 12.7243 25.9437 12.351C25.9132 12.2012 25.8078 11.9114 25.7095 11.707C25.4978 11.2669 25.0007 10.7602 24.5935 10.5694L24.3333 10.4475L24.3105 8.84797C24.2859 7.12328 24.278 7.06825 23.9665 6.45055C23.8188 6.1576 23.2932 5.62592 23.0036 5.4765C22.3735 5.15133 22.3762 5.15164 20.3264 5.12821L18.4381 5.10658L15.704 2.64231C14.2002 1.28698 12.9094 0.145155 12.8355 0.104972C12.6019 -0.0221545 12.3295 -0.0345383 12.1063 0.0717772ZM14.1822 3.57448L15.8964 5.11603H12.4292C10.4245 5.11603 8.98242 5.09692 9.0103 5.07071C9.49886 4.61189 12.4015 2.03293 12.4293 2.03293C12.4506 2.03293 13.2394 2.72663 14.1822 3.57448ZM22.2622 7.0369C22.5822 7.28376 22.5958 7.36027 22.5958 8.91687V10.3059H21.0951C19.9244 10.3059 19.5084 10.3238 19.203 10.3873C17.5501 10.731 16.2512 12.0571 15.9135 13.7457C15.8012 14.3074 15.8351 15.2012 15.9894 15.7439C16.4078 17.2157 17.5908 18.3613 19.0908 18.7471C19.4095 18.8291 19.7274 18.8496 21.036 18.8728L22.6002 18.9006L22.5853 20.329C22.5712 21.6789 22.5646 21.7652 22.4656 21.8992C22.408 21.9772 22.2978 22.0887 22.2206 22.147L22.0804 22.2529L12.9609 22.2662C3.92916 22.2793 3.83818 22.2784 3.50846 22.1754C2.70537 21.9245 2.09308 21.3176 1.81182 20.4938L1.69311 20.1461V14.5966C1.69311 9.35765 1.69834 9.02971 1.78642 8.73867C2.05371 7.85526 2.73941 7.18042 3.61193 6.942C3.83645 6.88064 5.27871 6.87093 12.9738 6.87915L22.0703 6.88881L22.2622 7.0369ZM23.7808 12.0717C23.974 12.1534 24.2048 12.4127 24.2593 12.6093C24.309 12.7885 24.309 16.3533 24.2593 16.5324C24.2048 16.729 23.974 16.9884 23.7808 17.07C23.6504 17.1251 23.1961 17.1401 21.6596 17.1401C19.7488 17.1401 19.6957 17.1374 19.3291 17.0196C18.6205 16.792 18.1019 16.3366 17.7823 15.6616C17.0985 14.2174 17.8613 12.5589 19.421 12.0988C19.6866 12.0205 19.9757 12.0084 21.6704 12.0053C23.2043 12.0024 23.65 12.0164 23.7808 12.0717ZM19.859 13.7509C19.2023 13.8999 18.9988 14.8129 19.5284 15.2343C19.8934 15.5248 20.3423 15.4846 20.6749 15.1318C20.9784 14.8099 20.9791 14.3277 20.6764 13.9894C20.4923 13.7836 20.1521 13.6843 19.859 13.7509Z", fill: "white" })));
};

var Icon$5 = function (props) {
    return (React.createElement(Svg, __assign({ viewBox: "0 0 27 24" }, props, { fill: "none" }),
        React.createElement("path", { opacity: "1", fillRule: "evenodd", clipRule: "evenodd", d: "M13.5106 0.0445072C11.8684 0.227097 9.93365 0.879675 8.56612 1.71209C6.42715 3.01412 4.57262 5.11359 3.70768 7.21218C3.57516 7.53381 3.44916 7.7973 3.42763 7.7978C3.40615 7.7983 3.19565 7.42555 2.95981 6.96958C2.72397 6.51361 2.46197 6.04869 2.37759 5.93655C2.02348 5.466 1.25243 5.31689 0.708659 5.6138C0.288101 5.84344 0.0575822 6.21069 0.0259088 6.70133L0 7.10317L1.44772 9.9044C2.24399 11.4451 2.96051 12.782 3.04001 12.8754C3.36852 13.261 4.06135 13.4143 4.55475 13.2107C4.68056 13.1588 5.84126 12.5206 7.1341 11.7926C9.71358 10.3399 9.95246 10.1649 10.0816 9.63327C10.3234 8.63779 9.33673 7.72753 8.37487 8.05874C8.25927 8.0986 7.62789 8.43474 6.97181 8.80586C6.2508 9.21372 5.77886 9.44668 5.77886 9.3948C5.77886 9.24676 6.14 8.33375 6.40277 7.81751C6.86387 6.91157 7.40858 6.18065 8.21366 5.38766C9.59024 4.03169 11.0971 3.23213 13.0653 2.8132C13.4842 2.72403 13.8806 2.69894 14.8691 2.69894C15.9641 2.69894 16.2253 2.71865 16.7936 2.84436C19.6985 3.48686 22.0159 5.26439 23.3307 7.85869C24.6385 10.4391 24.636 13.5521 23.3241 16.1306C22.0011 18.7309 19.7004 20.495 16.7936 21.1378C16.2253 21.2636 15.9641 21.2833 14.8691 21.2833C13.4539 21.2833 13.0203 21.2191 11.8629 20.8387C9.85579 20.1792 8.13808 18.8689 6.94976 17.0913C6.3595 16.2084 6.07121 15.9958 5.46403 15.9958C5.14229 15.9958 4.57528 16.2929 4.40101 16.5528C4.02391 17.115 4.06179 17.5815 4.54645 18.3441C7.01533 22.2285 11.4734 24.397 16.0501 23.9396C18.819 23.663 21.2572 22.5474 23.2793 20.6319C25.6909 18.3475 27 15.3073 27 11.9911C27 9.43567 26.2402 7.09485 24.7195 4.9651C24.2136 4.25652 22.848 2.8861 22.1223 2.35873C20.5675 1.22877 18.8057 0.481708 16.9279 0.156201C16.0819 0.00959092 14.3374 -0.0474759 13.5106 0.0445072ZM14.5357 5.357C14.1729 5.44197 13.851 5.70247 13.6733 6.05469L13.5072 6.38408L13.5086 9.32835C13.5098 12.0721 13.518 12.2898 13.6284 12.523C13.7068 12.6884 14.5063 13.5187 15.9867 14.9718C18.422 17.3622 18.4053 17.3496 19.0593 17.296C19.9687 17.2216 20.5334 16.2027 20.1096 15.4014C20.0346 15.2596 19.1648 14.3507 18.1041 13.3055L16.2311 11.46V9.03132C16.2311 7.37881 16.2073 6.50935 16.1566 6.31074C16.0545 5.90984 15.6094 5.47019 15.2035 5.36926C14.8765 5.28792 14.8339 5.2871 14.5357 5.357Z", fill: "white" })));
};

var Icon$6 = function (props) {
    return (React.createElement(Svg, __assign({ viewBox: "0 0 21 21" }, props, { fill: "none" }),
        React.createElement("g", { opacity: "1" },
            React.createElement("path", { fillRule: "evenodd", clipRule: "evenodd", d: "M8.19248 11.1925L6.07116 13.3138C5.68064 13.7043 5.68064 14.3375 6.07116 14.728C6.46169 15.1186 7.09485 15.1186 7.48538 14.728L9.6067 12.6067L10.3138 13.3138C11.0949 14.0949 11.0949 15.3612 10.3138 16.1422L7.48538 18.9707C6.70433 19.7517 5.438 19.7517 4.65695 18.9707L1.82852 16.1422C1.04747 15.3612 1.04747 14.0949 1.82852 13.3138L4.65695 10.4854C5.438 9.70434 6.70433 9.70434 7.48538 10.4854L8.19248 11.1925Z", fill: "white" }),
            React.createElement("path", { fillRule: "evenodd", clipRule: "evenodd", d: "M12.4349 9.77821L14.5563 7.65689C14.9468 7.26636 14.9468 6.6332 14.5563 6.24268C14.1657 5.85215 13.5326 5.85215 13.1421 6.24268L11.0207 8.364L10.3136 7.65689C9.53258 6.87584 9.53258 5.60951 10.3136 4.82846L13.1421 2.00004C13.9231 1.21899 15.1894 1.21899 15.9705 2.00004L18.7989 4.82846C19.58 5.60951 19.58 6.87584 18.7989 7.65689L15.9705 10.4853C15.1894 11.2664 13.9231 11.2664 13.1421 10.4853L12.4349 9.77821Z", fill: "white" }),
            React.createElement("rect", { x: "1.82837", y: "3.41406", width: "2", height: "4", rx: "1", transform: "rotate(-45 1.82837 3.41406)", fill: "white" }),
            React.createElement("rect", { x: "6.65674", width: "2", height: "4", rx: "1", fill: "white" }),
            React.createElement("rect", { width: "2", height: "4", rx: "1", transform: "matrix(0 1 1 0 0 6.65674)", fill: "white" }),
            React.createElement("rect", { width: "2", height: "4", rx: "1", transform: "matrix(0.707107 -0.707107 -0.707107 -0.707107 17.2632 18.8491)", fill: "white" }),
            React.createElement("rect", { width: "2", height: "4", rx: "1", transform: "matrix(0 -1 -1 0 20.6777 14.021)", fill: "white" }),
            React.createElement("rect", { x: "14.021", y: "20.6777", width: "2", height: "4", rx: "1", transform: "rotate(-180 14.021 20.6777)", fill: "white" }))));
};

var Icon$7 = function (props) {
    return (React.createElement(Svg, __assign({ viewBox: "0 0 24 24" }, props, { fill: "none" }),
        React.createElement("path", { d: "M17 21V19C17 17.9391 16.5786 16.9217 15.8284 16.1716C15.0783 15.4214 14.0609 15 13 15H5C3.93913 15 2.92172 15.4214 2.17157 16.1716C1.42143 16.9217 1 17.9391 1 19V21", stroke: "white", "stroke-width": "2", "stroke-linecap": "round", "stroke-linejoin": "round" }),
        React.createElement("path", { d: "M9 11C11.2091 11 13 9.20914 13 7C13 4.79086 11.2091 3 9 3C6.79086 3 5 4.79086 5 7C5 9.20914 6.79086 11 9 11Z", stroke: "white", "stroke-width": "2", "stroke-linecap": "round", "stroke-linejoin": "round" }),
        React.createElement("path", { d: "M23 20.9999V18.9999C22.9993 18.1136 22.7044 17.2527 22.1614 16.5522C21.6184 15.8517 20.8581 15.3515 20 15.1299", stroke: "white", "stroke-width": "2", "stroke-linecap": "round", "stroke-linejoin": "round" }),
        React.createElement("path", { d: "M16 3.12988C16.8604 3.35018 17.623 3.85058 18.1676 4.55219C18.7122 5.2538 19.0078 6.11671 19.0078 7.00488C19.0078 7.89305 18.7122 8.75596 18.1676 9.45757C17.623 10.1592 16.8604 10.6596 16 10.8799", stroke: "white", "stroke-width": "2", "stroke-linecap": "round", "stroke-linejoin": "round" })));
};

var Icon$8 = function (props) {
    return (React.createElement(Svg, __assign({ viewBox: "0 0 24 24", fill: '#fff' }, props),
        React.createElement("path", { d: "M8.71005 11.71L11.3001 14.3C11.6901 14.69 12.3201 14.69 12.7101 14.3L15.3001 11.71C15.9301 11.08 15.4801 10 14.5901 10H9.41005C8.52005 10 8.08005 11.08 8.71005 11.71Z" })));
};

var Icon$9 = function (props) {
    return (React.createElement(Svg, __assign({ viewBox: "0 0 26 26" }, props, { fill: "none" }),
        React.createElement("mask", { id: "path-1-outside-1_365_11596", maskUnits: "userSpaceOnUse", x: "0", y: "0", width: "26", height: "26", fill: "black" },
            React.createElement("rect", { fill: "white", width: "26", height: "26" }),
            React.createElement("path", { fillRule: "evenodd", clipRule: "evenodd", d: "M11.8991 1.02696C10.9402 1.11987 9.93841 1.34205 9.02306 1.66479C8.70181 1.77809 8.42341 1.88673 8.4044 1.90627C8.36145 1.95052 8.61887 2.55627 8.67468 2.54227C8.69702 2.53663 8.95819 2.44527 9.25507 2.33921C10.1937 2.00393 11.1832 1.79908 12.2944 1.71007L12.6347 1.68283V2.64058C12.6347 3.59416 12.6342 3.59848 12.529 3.62671C12.4709 3.64226 12.3495 3.65541 12.2591 3.65592C11.9849 3.65747 11.1839 3.79073 10.6867 3.91751C9.37458 4.25214 7.9706 4.97423 6.9961 5.81564C6.81567 5.9714 6.65317 6.09883 6.63496 6.09883C6.61674 6.09883 6.30028 5.79652 5.93172 5.42704L5.26162 4.7553L5.64333 4.42889C6.0761 4.0589 6.75442 3.56353 7.15457 3.32529C7.30299 3.23694 7.42447 3.14966 7.42447 3.13134C7.42447 3.09973 7.2989 2.86558 7.15832 2.6349L7.09406 2.52949L6.70771 2.77037C5.10415 3.77035 3.71936 5.16222 2.76528 6.73295C0.923833 9.76474 0.500212 13.417 1.6021 16.7614C2.5159 19.5349 4.37317 21.8439 6.90762 23.3571C8.28016 24.1766 9.87518 24.7209 11.5551 24.943C12.1302 25.019 13.8901 25.019 14.4653 24.943C17.388 24.5565 19.8655 23.2719 21.8874 21.0945C23.006 19.8898 23.8812 18.3915 24.4182 16.7614C24.8416 15.4765 25 14.4534 25 13.0037C25 11.9312 24.9548 11.4574 24.7624 10.5142C24.1028 7.28012 22.0264 4.39 19.1127 2.65031C17.7542 1.83915 16.1341 1.28333 14.5122 1.07187C13.9753 1.00188 12.4318 0.975387 11.8991 1.02696ZM14.2071 1.75686C16.3836 1.95019 18.6294 2.93735 20.4395 4.49625L20.8516 4.85107L20.1726 5.53047L19.4937 6.20987L19.2912 6.03612C18.8889 5.69093 18.3371 5.2788 17.9424 5.02867C16.8074 4.30935 15.2385 3.77866 13.9489 3.67786C13.3177 3.62854 13.3857 3.75348 13.3857 2.64213V1.67672L13.6321 1.70218C13.7676 1.71618 14.0264 1.74079 14.2071 1.75686ZM14.3224 4.43627C16.1786 4.72402 17.8004 5.54799 19.1306 6.87899C20.4608 8.21021 21.2861 9.83727 21.5711 11.6905C21.9598 14.2183 21.2361 16.7206 19.5431 18.7019C18.9661 19.3773 18.3863 19.8543 17.2431 20.5943C17.2256 20.6056 17.405 20.9649 17.5032 21.1153L17.5704 21.2183L17.8837 21.0266C18.1912 20.8384 18.8668 20.3424 19.2217 20.0443L19.4016 19.8932L20.0778 20.5702L20.754 21.2472L20.3672 21.5776C18.6044 23.0837 16.3244 24.0663 14.1602 24.2526C13.9795 24.2682 13.7313 24.2925 13.6086 24.3068L13.3857 24.3327V23.3675V22.4023L13.5382 22.3807C13.6222 22.3689 13.8984 22.336 14.1521 22.3078C14.7893 22.2368 15.2101 22.1443 15.9085 21.9215C16.2377 21.8165 16.512 21.7261 16.518 21.7206C16.5528 21.6891 16.2698 21.0828 16.2204 21.0828C16.1875 21.0828 16.0168 21.1344 15.841 21.1973C14.9431 21.5189 14.0321 21.6655 12.9632 21.6602C10.6272 21.6488 8.53827 20.778 6.88979 19.1284C5.55541 17.793 4.76008 16.2234 4.43996 14.2934C4.3352 13.6615 4.3352 12.3458 4.43996 11.714C4.86917 9.12621 6.26996 6.99811 8.43721 5.64119C10.1749 4.55323 12.2872 4.12076 14.3224 4.43627ZM6.11018 6.62398C6.11018 6.6422 5.98284 6.80482 5.82719 6.98538C4.98638 7.96056 4.26479 9.36553 3.9304 10.6786C3.80371 11.1761 3.67054 11.9777 3.669 12.2521C3.66848 12.3426 3.65534 12.4641 3.6398 12.5222C3.61159 12.6274 3.60732 12.6279 2.6524 12.6279H1.6933L1.72137 12.3343C1.81961 11.3075 1.87434 10.9753 2.07655 10.1788C2.49135 8.54489 3.31902 6.95752 4.44231 5.64161L4.77243 5.25484L5.44131 5.92283C5.80921 6.29024 6.11018 6.60575 6.11018 6.62398ZM21.6234 5.68637C23.2351 7.6559 24.0986 9.78846 24.2961 12.2874L24.323 12.6279H23.366C22.2851 12.6279 22.3531 12.6543 22.3516 12.2342C22.3498 11.7293 22.1076 10.617 21.8237 9.8096C21.5084 8.91296 20.8205 7.71034 20.2385 7.03803C20.1096 6.88904 20.004 6.74916 20.004 6.72717C20.004 6.66841 21.2852 5.39379 21.3412 5.39693C21.3673 5.39839 21.4943 5.52864 21.6234 5.68637ZM12.6347 5.84049V6.19278H13.0102H13.3857V5.84049V5.4882H13.0102H12.6347V5.84049ZM7.37753 7.71936V8.07165H7.75304H8.12855V7.71936V7.36707H7.75304H7.37753V7.71936ZM18.2673 8.09513V8.44742H18.6428H19.0183V8.09513V7.74284H18.6428H18.2673V8.09513ZM11.1178 10.4239C10.0901 11.7953 9.2394 12.9429 9.22738 12.9743C9.20391 13.0354 12.9412 18.0531 13.0102 18.0531C13.0525 18.0531 16.8122 13.0598 16.8122 13.0037C16.8122 12.9623 13.0617 7.95764 13.0193 7.94243C13.0012 7.93594 12.1456 9.05265 11.1178 10.4239ZM14.4861 11.0422C15.275 12.0948 15.9204 12.9774 15.9204 13.0037C15.9204 13.0704 13.0603 16.8789 13.0102 16.8789C12.9446 16.8789 10.0929 13.0501 10.1089 12.9837C10.1313 12.8908 12.9815 9.10319 13.0189 9.11663C13.037 9.12316 13.6973 9.98969 14.4861 11.0422ZM5.49998 13.0037V13.3834L5.86375 13.3697L6.22753 13.356V13.0037V12.6514L5.86375 12.6377L5.49998 12.624V13.0037ZM19.7889 12.6916C19.7755 12.7266 19.7709 12.8904 19.7787 13.0556L19.7928 13.356L20.1566 13.3697L20.5204 13.3834V13.0057V12.6279H20.1669C19.9127 12.6279 19.8065 12.6458 19.7889 12.6916ZM3.64205 13.4851C3.65379 13.5433 3.6758 13.7494 3.69092 13.9431C3.79165 15.2336 4.32196 16.8036 5.04078 17.9394C5.29073 18.3344 5.70257 18.8865 6.04752 19.2892L6.22115 19.4918L5.54222 20.1712L4.8633 20.8506L4.50873 20.4383C2.95093 18.6269 1.96446 16.3794 1.77126 14.2015C1.75521 14.0206 1.73062 13.7617 1.71663 13.6261L1.69119 13.3795H2.65592C3.61872 13.3795 3.6207 13.3796 3.64205 13.4851ZM24.3053 13.6026C24.2911 13.7253 24.2667 13.9736 24.2511 14.1545C24.065 16.3203 23.0824 18.6034 21.578 20.3658L21.2479 20.7525L20.579 20.0845C20.2112 19.7171 19.9102 19.4055 19.9102 19.3919C19.9102 19.3784 20.047 19.2032 20.2143 19.0026C21.0465 18.005 21.6832 16.784 22.0476 15.4871C22.1839 15.002 22.3507 14.0686 22.3516 13.7858C22.353 13.351 22.2818 13.3795 23.37 13.3795H24.3312L24.3053 13.6026ZM7.00202 17.9122V18.2645H7.37753H7.75304V17.9122V17.5599H7.37753H7.00202V17.9122ZM18.2673 17.9122V18.2645H18.6428H19.0183V17.9122V17.5599H18.6428H18.2673V17.9122ZM12.6347 20.1669V20.5192H13.0102H13.3857V20.1669V19.8146H13.0102H12.6347V20.1669ZM7.04872 20.2371C7.72055 20.8195 8.92233 21.5079 9.81834 21.8234C10.6251 22.1076 11.7367 22.3499 12.2413 22.3517C12.6611 22.3532 12.6347 22.2846 12.6347 23.3708V24.3327L12.4117 24.3068C12.289 24.2925 12.0409 24.2682 11.8602 24.2526C10.356 24.1231 8.66008 23.5589 7.2253 22.7107C6.73128 22.4186 5.96486 21.8608 5.61733 21.5402L5.38263 21.3238L6.03916 20.6631C6.40022 20.2998 6.71419 20.0025 6.73686 20.0025C6.75953 20.0025 6.89983 20.1081 7.04872 20.2371Z" })),
        React.createElement("path", { fillRule: "evenodd", clipRule: "evenodd", d: "M11.8991 1.02696C10.9402 1.11987 9.93841 1.34205 9.02306 1.66479C8.70181 1.77809 8.42341 1.88673 8.4044 1.90627C8.36145 1.95052 8.61887 2.55627 8.67468 2.54227C8.69702 2.53663 8.95819 2.44527 9.25507 2.33921C10.1937 2.00393 11.1832 1.79908 12.2944 1.71007L12.6347 1.68283V2.64058C12.6347 3.59416 12.6342 3.59848 12.529 3.62671C12.4709 3.64226 12.3495 3.65541 12.2591 3.65592C11.9849 3.65747 11.1839 3.79073 10.6867 3.91751C9.37458 4.25214 7.9706 4.97423 6.9961 5.81564C6.81567 5.9714 6.65317 6.09883 6.63496 6.09883C6.61674 6.09883 6.30028 5.79652 5.93172 5.42704L5.26162 4.7553L5.64333 4.42889C6.0761 4.0589 6.75442 3.56353 7.15457 3.32529C7.30299 3.23694 7.42447 3.14966 7.42447 3.13134C7.42447 3.09973 7.2989 2.86558 7.15832 2.6349L7.09406 2.52949L6.70771 2.77037C5.10415 3.77035 3.71936 5.16222 2.76528 6.73295C0.923833 9.76474 0.500212 13.417 1.6021 16.7614C2.5159 19.5349 4.37317 21.8439 6.90762 23.3571C8.28016 24.1766 9.87518 24.7209 11.5551 24.943C12.1302 25.019 13.8901 25.019 14.4653 24.943C17.388 24.5565 19.8655 23.2719 21.8874 21.0945C23.006 19.8898 23.8812 18.3915 24.4182 16.7614C24.8416 15.4765 25 14.4534 25 13.0037C25 11.9312 24.9548 11.4574 24.7624 10.5142C24.1028 7.28012 22.0264 4.39 19.1127 2.65031C17.7542 1.83915 16.1341 1.28333 14.5122 1.07187C13.9753 1.00188 12.4318 0.975387 11.8991 1.02696ZM14.2071 1.75686C16.3836 1.95019 18.6294 2.93735 20.4395 4.49625L20.8516 4.85107L20.1726 5.53047L19.4937 6.20987L19.2912 6.03612C18.8889 5.69093 18.3371 5.2788 17.9424 5.02867C16.8074 4.30935 15.2385 3.77866 13.9489 3.67786C13.3177 3.62854 13.3857 3.75348 13.3857 2.64213V1.67672L13.6321 1.70218C13.7676 1.71618 14.0264 1.74079 14.2071 1.75686ZM14.3224 4.43627C16.1786 4.72402 17.8004 5.54799 19.1306 6.87899C20.4608 8.21021 21.2861 9.83727 21.5711 11.6905C21.9598 14.2183 21.2361 16.7206 19.5431 18.7019C18.9661 19.3773 18.3863 19.8543 17.2431 20.5943C17.2256 20.6056 17.405 20.9649 17.5032 21.1153L17.5704 21.2183L17.8837 21.0266C18.1912 20.8384 18.8668 20.3424 19.2217 20.0443L19.4016 19.8932L20.0778 20.5702L20.754 21.2472L20.3672 21.5776C18.6044 23.0837 16.3244 24.0663 14.1602 24.2526C13.9795 24.2682 13.7313 24.2925 13.6086 24.3068L13.3857 24.3327V23.3675V22.4023L13.5382 22.3807C13.6222 22.3689 13.8984 22.336 14.1521 22.3078C14.7893 22.2368 15.2101 22.1443 15.9085 21.9215C16.2377 21.8165 16.512 21.7261 16.518 21.7206C16.5528 21.6891 16.2698 21.0828 16.2204 21.0828C16.1875 21.0828 16.0168 21.1344 15.841 21.1973C14.9431 21.5189 14.0321 21.6655 12.9632 21.6602C10.6272 21.6488 8.53827 20.778 6.88979 19.1284C5.55541 17.793 4.76008 16.2234 4.43996 14.2934C4.3352 13.6615 4.3352 12.3458 4.43996 11.714C4.86917 9.12621 6.26996 6.99811 8.43721 5.64119C10.1749 4.55323 12.2872 4.12076 14.3224 4.43627ZM6.11018 6.62398C6.11018 6.6422 5.98284 6.80482 5.82719 6.98538C4.98638 7.96056 4.26479 9.36553 3.9304 10.6786C3.80371 11.1761 3.67054 11.9777 3.669 12.2521C3.66848 12.3426 3.65534 12.4641 3.6398 12.5222C3.61159 12.6274 3.60732 12.6279 2.6524 12.6279H1.6933L1.72137 12.3343C1.81961 11.3075 1.87434 10.9753 2.07655 10.1788C2.49135 8.54489 3.31902 6.95752 4.44231 5.64161L4.77243 5.25484L5.44131 5.92283C5.80921 6.29024 6.11018 6.60575 6.11018 6.62398ZM21.6234 5.68637C23.2351 7.6559 24.0986 9.78846 24.2961 12.2874L24.323 12.6279H23.366C22.2851 12.6279 22.3531 12.6543 22.3516 12.2342C22.3498 11.7293 22.1076 10.617 21.8237 9.8096C21.5084 8.91296 20.8205 7.71034 20.2385 7.03803C20.1096 6.88904 20.004 6.74916 20.004 6.72717C20.004 6.66841 21.2852 5.39379 21.3412 5.39693C21.3673 5.39839 21.4943 5.52864 21.6234 5.68637ZM12.6347 5.84049V6.19278H13.0102H13.3857V5.84049V5.4882H13.0102H12.6347V5.84049ZM7.37753 7.71936V8.07165H7.75304H8.12855V7.71936V7.36707H7.75304H7.37753V7.71936ZM18.2673 8.09513V8.44742H18.6428H19.0183V8.09513V7.74284H18.6428H18.2673V8.09513ZM11.1178 10.4239C10.0901 11.7953 9.2394 12.9429 9.22738 12.9743C9.20391 13.0354 12.9412 18.0531 13.0102 18.0531C13.0525 18.0531 16.8122 13.0598 16.8122 13.0037C16.8122 12.9623 13.0617 7.95764 13.0193 7.94243C13.0012 7.93594 12.1456 9.05265 11.1178 10.4239ZM14.4861 11.0422C15.275 12.0948 15.9204 12.9774 15.9204 13.0037C15.9204 13.0704 13.0603 16.8789 13.0102 16.8789C12.9446 16.8789 10.0929 13.0501 10.1089 12.9837C10.1313 12.8908 12.9815 9.10319 13.0189 9.11663C13.037 9.12316 13.6973 9.98969 14.4861 11.0422ZM5.49998 13.0037V13.3834L5.86375 13.3697L6.22753 13.356V13.0037V12.6514L5.86375 12.6377L5.49998 12.624V13.0037ZM19.7889 12.6916C19.7755 12.7266 19.7709 12.8904 19.7787 13.0556L19.7928 13.356L20.1566 13.3697L20.5204 13.3834V13.0057V12.6279H20.1669C19.9127 12.6279 19.8065 12.6458 19.7889 12.6916ZM3.64205 13.4851C3.65379 13.5433 3.6758 13.7494 3.69092 13.9431C3.79165 15.2336 4.32196 16.8036 5.04078 17.9394C5.29073 18.3344 5.70257 18.8865 6.04752 19.2892L6.22115 19.4918L5.54222 20.1712L4.8633 20.8506L4.50873 20.4383C2.95093 18.6269 1.96446 16.3794 1.77126 14.2015C1.75521 14.0206 1.73062 13.7617 1.71663 13.6261L1.69119 13.3795H2.65592C3.61872 13.3795 3.6207 13.3796 3.64205 13.4851ZM24.3053 13.6026C24.2911 13.7253 24.2667 13.9736 24.2511 14.1545C24.065 16.3203 23.0824 18.6034 21.578 20.3658L21.2479 20.7525L20.579 20.0845C20.2112 19.7171 19.9102 19.4055 19.9102 19.3919C19.9102 19.3784 20.047 19.2032 20.2143 19.0026C21.0465 18.005 21.6832 16.784 22.0476 15.4871C22.1839 15.002 22.3507 14.0686 22.3516 13.7858C22.353 13.351 22.2818 13.3795 23.37 13.3795H24.3312L24.3053 13.6026ZM7.00202 17.9122V18.2645H7.37753H7.75304V17.9122V17.5599H7.37753H7.00202V17.9122ZM18.2673 17.9122V18.2645H18.6428H19.0183V17.9122V17.5599H18.6428H18.2673V17.9122ZM12.6347 20.1669V20.5192H13.0102H13.3857V20.1669V19.8146H13.0102H12.6347V20.1669ZM7.04872 20.2371C7.72055 20.8195 8.92233 21.5079 9.81834 21.8234C10.6251 22.1076 11.7367 22.3499 12.2413 22.3517C12.6611 22.3532 12.6347 22.2846 12.6347 23.3708V24.3327L12.4117 24.3068C12.289 24.2925 12.0409 24.2682 11.8602 24.2526C10.356 24.1231 8.66008 23.5589 7.2253 22.7107C6.73128 22.4186 5.96486 21.8608 5.61733 21.5402L5.38263 21.3238L6.03916 20.6631C6.40022 20.2998 6.71419 20.0025 6.73686 20.0025C6.75953 20.0025 6.89983 20.1081 7.04872 20.2371Z", fill: "white" }),
        React.createElement("path", { fillRule: "evenodd", clipRule: "evenodd", d: "M11.8991 1.02696C10.9402 1.11987 9.93841 1.34205 9.02306 1.66479C8.70181 1.77809 8.42341 1.88673 8.4044 1.90627C8.36145 1.95052 8.61887 2.55627 8.67468 2.54227C8.69702 2.53663 8.95819 2.44527 9.25507 2.33921C10.1937 2.00393 11.1832 1.79908 12.2944 1.71007L12.6347 1.68283V2.64058C12.6347 3.59416 12.6342 3.59848 12.529 3.62671C12.4709 3.64226 12.3495 3.65541 12.2591 3.65592C11.9849 3.65747 11.1839 3.79073 10.6867 3.91751C9.37458 4.25214 7.9706 4.97423 6.9961 5.81564C6.81567 5.9714 6.65317 6.09883 6.63496 6.09883C6.61674 6.09883 6.30028 5.79652 5.93172 5.42704L5.26162 4.7553L5.64333 4.42889C6.0761 4.0589 6.75442 3.56353 7.15457 3.32529C7.30299 3.23694 7.42447 3.14966 7.42447 3.13134C7.42447 3.09973 7.2989 2.86558 7.15832 2.6349L7.09406 2.52949L6.70771 2.77037C5.10415 3.77035 3.71936 5.16222 2.76528 6.73295C0.923833 9.76474 0.500212 13.417 1.6021 16.7614C2.5159 19.5349 4.37317 21.8439 6.90762 23.3571C8.28016 24.1766 9.87518 24.7209 11.5551 24.943C12.1302 25.019 13.8901 25.019 14.4653 24.943C17.388 24.5565 19.8655 23.2719 21.8874 21.0945C23.006 19.8898 23.8812 18.3915 24.4182 16.7614C24.8416 15.4765 25 14.4534 25 13.0037C25 11.9312 24.9548 11.4574 24.7624 10.5142C24.1028 7.28012 22.0264 4.39 19.1127 2.65031C17.7542 1.83915 16.1341 1.28333 14.5122 1.07187C13.9753 1.00188 12.4318 0.975387 11.8991 1.02696ZM14.2071 1.75686C16.3836 1.95019 18.6294 2.93735 20.4395 4.49625L20.8516 4.85107L20.1726 5.53047L19.4937 6.20987L19.2912 6.03612C18.8889 5.69093 18.3371 5.2788 17.9424 5.02867C16.8074 4.30935 15.2385 3.77866 13.9489 3.67786C13.3177 3.62854 13.3857 3.75348 13.3857 2.64213V1.67672L13.6321 1.70218C13.7676 1.71618 14.0264 1.74079 14.2071 1.75686ZM14.3224 4.43627C16.1786 4.72402 17.8004 5.54799 19.1306 6.87899C20.4608 8.21021 21.2861 9.83727 21.5711 11.6905C21.9598 14.2183 21.2361 16.7206 19.5431 18.7019C18.9661 19.3773 18.3863 19.8543 17.2431 20.5943C17.2256 20.6056 17.405 20.9649 17.5032 21.1153L17.5704 21.2183L17.8837 21.0266C18.1912 20.8384 18.8668 20.3424 19.2217 20.0443L19.4016 19.8932L20.0778 20.5702L20.754 21.2472L20.3672 21.5776C18.6044 23.0837 16.3244 24.0663 14.1602 24.2526C13.9795 24.2682 13.7313 24.2925 13.6086 24.3068L13.3857 24.3327V23.3675V22.4023L13.5382 22.3807C13.6222 22.3689 13.8984 22.336 14.1521 22.3078C14.7893 22.2368 15.2101 22.1443 15.9085 21.9215C16.2377 21.8165 16.512 21.7261 16.518 21.7206C16.5528 21.6891 16.2698 21.0828 16.2204 21.0828C16.1875 21.0828 16.0168 21.1344 15.841 21.1973C14.9431 21.5189 14.0321 21.6655 12.9632 21.6602C10.6272 21.6488 8.53827 20.778 6.88979 19.1284C5.55541 17.793 4.76008 16.2234 4.43996 14.2934C4.3352 13.6615 4.3352 12.3458 4.43996 11.714C4.86917 9.12621 6.26996 6.99811 8.43721 5.64119C10.1749 4.55323 12.2872 4.12076 14.3224 4.43627ZM6.11018 6.62398C6.11018 6.6422 5.98284 6.80482 5.82719 6.98538C4.98638 7.96056 4.26479 9.36553 3.9304 10.6786C3.80371 11.1761 3.67054 11.9777 3.669 12.2521C3.66848 12.3426 3.65534 12.4641 3.6398 12.5222C3.61159 12.6274 3.60732 12.6279 2.6524 12.6279H1.6933L1.72137 12.3343C1.81961 11.3075 1.87434 10.9753 2.07655 10.1788C2.49135 8.54489 3.31902 6.95752 4.44231 5.64161L4.77243 5.25484L5.44131 5.92283C5.80921 6.29024 6.11018 6.60575 6.11018 6.62398ZM21.6234 5.68637C23.2351 7.6559 24.0986 9.78846 24.2961 12.2874L24.323 12.6279H23.366C22.2851 12.6279 22.3531 12.6543 22.3516 12.2342C22.3498 11.7293 22.1076 10.617 21.8237 9.8096C21.5084 8.91296 20.8205 7.71034 20.2385 7.03803C20.1096 6.88904 20.004 6.74916 20.004 6.72717C20.004 6.66841 21.2852 5.39379 21.3412 5.39693C21.3673 5.39839 21.4943 5.52864 21.6234 5.68637ZM12.6347 5.84049V6.19278H13.0102H13.3857V5.84049V5.4882H13.0102H12.6347V5.84049ZM7.37753 7.71936V8.07165H7.75304H8.12855V7.71936V7.36707H7.75304H7.37753V7.71936ZM18.2673 8.09513V8.44742H18.6428H19.0183V8.09513V7.74284H18.6428H18.2673V8.09513ZM11.1178 10.4239C10.0901 11.7953 9.2394 12.9429 9.22738 12.9743C9.20391 13.0354 12.9412 18.0531 13.0102 18.0531C13.0525 18.0531 16.8122 13.0598 16.8122 13.0037C16.8122 12.9623 13.0617 7.95764 13.0193 7.94243C13.0012 7.93594 12.1456 9.05265 11.1178 10.4239ZM14.4861 11.0422C15.275 12.0948 15.9204 12.9774 15.9204 13.0037C15.9204 13.0704 13.0603 16.8789 13.0102 16.8789C12.9446 16.8789 10.0929 13.0501 10.1089 12.9837C10.1313 12.8908 12.9815 9.10319 13.0189 9.11663C13.037 9.12316 13.6973 9.98969 14.4861 11.0422ZM5.49998 13.0037V13.3834L5.86375 13.3697L6.22753 13.356V13.0037V12.6514L5.86375 12.6377L5.49998 12.624V13.0037ZM19.7889 12.6916C19.7755 12.7266 19.7709 12.8904 19.7787 13.0556L19.7928 13.356L20.1566 13.3697L20.5204 13.3834V13.0057V12.6279H20.1669C19.9127 12.6279 19.8065 12.6458 19.7889 12.6916ZM3.64205 13.4851C3.65379 13.5433 3.6758 13.7494 3.69092 13.9431C3.79165 15.2336 4.32196 16.8036 5.04078 17.9394C5.29073 18.3344 5.70257 18.8865 6.04752 19.2892L6.22115 19.4918L5.54222 20.1712L4.8633 20.8506L4.50873 20.4383C2.95093 18.6269 1.96446 16.3794 1.77126 14.2015C1.75521 14.0206 1.73062 13.7617 1.71663 13.6261L1.69119 13.3795H2.65592C3.61872 13.3795 3.6207 13.3796 3.64205 13.4851ZM24.3053 13.6026C24.2911 13.7253 24.2667 13.9736 24.2511 14.1545C24.065 16.3203 23.0824 18.6034 21.578 20.3658L21.2479 20.7525L20.579 20.0845C20.2112 19.7171 19.9102 19.4055 19.9102 19.3919C19.9102 19.3784 20.047 19.2032 20.2143 19.0026C21.0465 18.005 21.6832 16.784 22.0476 15.4871C22.1839 15.002 22.3507 14.0686 22.3516 13.7858C22.353 13.351 22.2818 13.3795 23.37 13.3795H24.3312L24.3053 13.6026ZM7.00202 17.9122V18.2645H7.37753H7.75304V17.9122V17.5599H7.37753H7.00202V17.9122ZM18.2673 17.9122V18.2645H18.6428H19.0183V17.9122V17.5599H18.6428H18.2673V17.9122ZM12.6347 20.1669V20.5192H13.0102H13.3857V20.1669V19.8146H13.0102H12.6347V20.1669ZM7.04872 20.2371C7.72055 20.8195 8.92233 21.5079 9.81834 21.8234C10.6251 22.1076 11.7367 22.3499 12.2413 22.3517C12.6611 22.3532 12.6347 22.2846 12.6347 23.3708V24.3327L12.4117 24.3068C12.289 24.2925 12.0409 24.2682 11.8602 24.2526C10.356 24.1231 8.66008 23.5589 7.2253 22.7107C6.73128 22.4186 5.96486 21.8608 5.61733 21.5402L5.38263 21.3238L6.03916 20.6631C6.40022 20.2998 6.71419 20.0025 6.73686 20.0025C6.75953 20.0025 6.89983 20.1081 7.04872 20.2371Z", stroke: "white", strokeWidth: "0.6", mask: "url(#path-1-outside-1_365_11596)" })));
};

var TooltipText = styled(Text)(templateObject_1$b || (templateObject_1$b = __makeTemplateObject(["\n  text-decoration: ", ";\n  text-underline-offset: 0.2em;\n"], ["\n  text-decoration: ", ";\n  text-underline-offset: 0.2em;\n"])), function (_a) {
    var theme = _a.theme;
    return "underline dotted " + theme;
});
var templateObject_1$b;

var Tds = styled.td(templateObject_1$c || (templateObject_1$c = __makeTemplateObject(["\n  color: #464E5F;\n  opacity: 0.8;\n  padding: 10px;\n  vertical-align: middle;\n\n  ", "\n"], ["\n  color: #464E5F;\n  opacity: 0.8;\n  padding: 10px;\n  vertical-align: middle;\n\n  ", "\n"])), typography);
var Ths = styled(Tds).attrs({ as: "th" })(templateObject_2$2 || (templateObject_2$2 = __makeTemplateObject(["\n  color: #B5B5C3;\n\n  opacity: 0.8;\n  font-size: 12px;\n  text-transform: uppercase;\n"], ["\n  color: #B5B5C3;\n\n  opacity: 0.8;\n  font-size: 12px;\n  text-transform: uppercase;\n"])));
var templateObject_1$c, templateObject_2$2;

var Table = styled.table(templateObject_1$d || (templateObject_1$d = __makeTemplateObject(["\n  max-width: 100%;\n  width: 100%;\n\n  tbody tr:last-child {\n    ", " {\n      border-bottom: 0;\n    }\n  }\n\n  ", "\n"], ["\n  max-width: 100%;\n  width: 100%;\n\n  tbody tr:last-child {\n    ", " {\n      border-bottom: 0;\n    }\n  }\n\n  ", "\n"])), Tds, space);
var templateObject_1$d;

var byTextAscending = function (getTextProperty) {
    return function (objectA, objectB) {
        var upperA = getTextProperty(objectA).toUpperCase();
        var upperB = getTextProperty(objectB).toUpperCase();
        if (upperA < upperB) {
            return -1;
        }
        if (upperA > upperB) {
            return 1;
        }
        return 0;
    };
};
var byTextDescending = function (getTextProperty) {
    return function (objectA, objectB) {
        var upperA = getTextProperty(objectA).toUpperCase();
        var upperB = getTextProperty(objectB).toUpperCase();
        if (upperA > upperB) {
            return -1;
        }
        if (upperA < upperB) {
            return 1;
        }
        return 0;
    };
};

var sortByColumn = function (data, sortColumn, columns) {
    var isAscending = null;
    var sortedRows = __spreadArrays(data);
    columns.forEach(function (column) {
        // if the row was found
        if (sortColumn === column.name) {
            isAscending = column.sorted.asc;
            if (column.sort) {
                sortedRows = isAscending ? data.sort(column.sort) : data.sort(column.sort).reverse();
                // default to sort by string
            }
            else {
                sortedRows = isAscending
                    ? data.sort(byTextAscending(function (object) { return object.original[sortColumn]; }))
                    : data.sort(byTextDescending(function (object) { return object.original[sortColumn]; }));
            }
        }
    });
    return sortedRows;
};
var getPaginatedData = function (rows, perPage, page) {
    var start = (page - 1) * perPage;
    var end = start + perPage;
    return rows.slice(start, end);
};
var getColumnsByName = function (columns) {
    var columnsByName = {};
    columns.forEach(function (column) {
        var col = {
            name: column.name,
            label: column.label,
        };
        if (column.render) {
            col.render = column.render;
        }
        col.hidden = column.hidden;
        columnsByName[column.name] = col;
    });
    return columnsByName;
};
var createReducer = function () {
    return function (state, action) {
        var rows = [];
        var nextPage = 0;
        var prevPage = 0;
        var isAscending = null;
        var sortedRows = [];
        var columnCopy = [];
        var filteredRows = [];
        var selectedRowsById = {};
        var stateCopy = __assign({}, state);
        var rowIds = {};
        switch (action.type) {
            case "SET_ROWS":
                rows = __spreadArrays(action.data);
                // preserve sorting if a sort is already enabled when data changes
                if (state.sortColumn) {
                    rows = sortByColumn(action.data, state.sortColumn, state.columns);
                }
                if (state.paginationEnabled === true) {
                    rows = getPaginatedData(rows, state.pagination.perPage, state.pagination.page);
                }
                if (state.paginationEnabled === true) {
                    rows = getPaginatedData(rows, state.pagination.perPage, state.pagination.page);
                }
                columnCopy = state.columns.map(function (column) {
                    if (state.sortColumn === column.name) {
                        return __assign(__assign({}, column), { sorted: {
                                on: true,
                                asc: column.sorted.asc,
                            } });
                    }
                    return column;
                });
                return __assign(__assign({}, state), { rows: rows, originalRows: action.data, columns: columnCopy });
            case "NEXT_PAGE":
                nextPage = state.pagination.page + 1;
                return __assign(__assign({}, state), { rows: getPaginatedData(state.originalRows, state.pagination.perPage, nextPage), pagination: __assign(__assign({}, state.pagination), { page: nextPage, canNext: nextPage * state.pagination.perPage < state.originalRows.length, canPrev: nextPage !== 1 }) });
            case "PREV_PAGE":
                prevPage = state.pagination.page === 1 ? 1 : state.pagination.page - 1;
                return __assign(__assign({}, state), { rows: getPaginatedData(state.originalRows, state.pagination.perPage, prevPage), pagination: __assign(__assign({}, state.pagination), { page: prevPage, canNext: prevPage * state.pagination.perPage < state.originalRows.length, canPrev: prevPage !== 1 }) });
            case "TOGGLE_SORT":
                if (!(action.columnName in state.columnsByName)) {
                    throw new Error("Invalid column, " + action.columnName + " not found");
                }
                // loop through all columns and set the sort parameter to off unless
                // it's the specified column (only one column at a time for )
                columnCopy = state.columns.map(function (column) {
                    // if the row was found
                    if (action.columnName === column.name) {
                        if (action.isAscOverride !== undefined) {
                            // force the sort order
                            isAscending = action.isAscOverride;
                        }
                        else {
                            // if it's undefined, start by setting to ascending, otherwise toggle
                            isAscending = column.sorted.asc === undefined ? true : !column.sorted.asc;
                        }
                        if (column.sort) {
                            sortedRows = isAscending ? state.rows.sort(column.sort) : state.rows.sort(column.sort).reverse();
                            // default to sort by string
                        }
                        else {
                            sortedRows = isAscending
                                ? state.rows.sort(byTextAscending(function (object) { return object.original[action.columnName]; }))
                                : state.rows.sort(byTextDescending(function (object) { return object.original[action.columnName]; }));
                        }
                        return __assign(__assign({}, column), { sorted: {
                                on: true,
                                asc: isAscending,
                            } });
                    }
                    // set sorting to false for all other columns
                    return __assign(__assign({}, column), { sorted: {
                            on: false,
                            asc: false,
                        } });
                });
                return __assign(__assign({}, state), { columns: columnCopy, rows: sortedRows, sortColumn: action.columnName, columnsByName: getColumnsByName(columnCopy) });
            case "GLOBAL_FILTER":
                filteredRows = action.filter(state.originalRows);
                selectedRowsById = {};
                state.selectedRows.forEach(function (row) {
                    var _a;
                    selectedRowsById[row.id] = (_a = row.selected) !== null && _a !== void 0 ? _a : false;
                });
                return __assign(__assign({}, state), { rows: filteredRows.map(function (row) {
                        return selectedRowsById[row.id] ? __assign(__assign({}, row), { selected: selectedRowsById[row.id] }) : __assign({}, row);
                    }), filterOn: true });
            case "SELECT_ROW":
                stateCopy = __assign({}, state);
                stateCopy.rows = stateCopy.rows.map(function (row) {
                    var newRow = __assign({}, row);
                    if (newRow.id === action.rowId) {
                        newRow.selected = !newRow.selected;
                    }
                    return newRow;
                });
                stateCopy.originalRows = stateCopy.originalRows.map(function (row) {
                    var newRow = __assign({}, row);
                    if (newRow.id === action.rowId) {
                        newRow.selected = !newRow.selected;
                    }
                    return newRow;
                });
                stateCopy.selectedRows = stateCopy.originalRows.filter(function (row) { return row.selected === true; });
                stateCopy.toggleAllState =
                    stateCopy.selectedRows.length === stateCopy.rows.length
                        ? (stateCopy.toggleAllState = true)
                        : (stateCopy.toggleAllState = false);
                return stateCopy;
            case "SEARCH_STRING":
                stateCopy = __assign({}, state);
                stateCopy.rows = stateCopy.originalRows.filter(function (row) {
                    return (row.cells.filter(function (cell) {
                        if (cell.value.includes(action.searchString)) {
                            return true;
                        }
                        return false;
                    }).length > 0);
                });
                return stateCopy;
            case "TOGGLE_ALL":
                if (state.selectedRows.length < state.rows.length) {
                    stateCopy.rows = stateCopy.rows.map(function (row) {
                        rowIds[row.id] = true;
                        return __assign(__assign({}, row), { selected: true });
                    });
                    stateCopy.toggleAllState = true;
                }
                else {
                    stateCopy.rows = stateCopy.rows.map(function (row) {
                        rowIds[row.id] = false;
                        return __assign(__assign({}, row), { selected: false });
                    });
                    stateCopy.toggleAllState = false;
                }
                stateCopy.originalRows = stateCopy.originalRows.map(function (row) {
                    return row.id in rowIds ? __assign(__assign({}, row), { selected: rowIds[row.id] }) : __assign({}, row);
                });
                stateCopy.selectedRows = stateCopy.originalRows.filter(function (row) { return row.selected; });
                return stateCopy;
            default:
                throw new Error("Invalid reducer action");
        }
    };
};
var sortDataInOrder = function (data, columns) {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    return data.map(function (row) {
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        var newRow = {};
        columns.forEach(function (column) {
            if (!(column.name in row)) {
                throw new Error("Invalid row data, " + column.name + " not found");
            }
            newRow[column.name] = row[column.name];
        });
        return newRow;
    });
};
var makeRender = function (
// eslint-disable-next-line
value, 
// eslint-disable-next-line @typescript-eslint/no-explicit-any
render, row) {
    return render ? function () { return render({ row: row, value: value }); } : function () { return value; };
};
var makeHeaderRender = function (label, render) {
    return render ? function () { return render({ label: label }); } : function () { return label; };
};
var useTable = function (columns, data, options) {
    var columnsWithSorting = useMemo(function () {
        return columns.map(function (column) {
            return __assign(__assign({}, column), { label: column.label ? column.label : column.name, hidden: column.hidden ? column.hidden : false, sort: column.sort, sorted: {
                    on: false,
                    asc: false,
                } });
        });
    }, [columns]);
    var columnsByName = useMemo(function () { return getColumnsByName(columnsWithSorting); }, [columnsWithSorting]);
    var tableData = useMemo(function () {
        var sortedData = sortDataInOrder(data, columnsWithSorting);
        var newData = sortedData.map(function (row, idx) {
            return {
                id: idx,
                selected: false,
                hidden: false,
                original: row,
                cells: Object.entries(row)
                    .map(function (_a) {
                    var column = _a[0], value = _a[1];
                    return {
                        hidden: columnsByName[column].hidden,
                        field: column,
                        value: value,
                        render: makeRender(value, columnsByName[column].render, row),
                    };
                })
                    .filter(function (cell) { return !cell.hidden; }),
            };
        });
        return newData;
    }, [data, columnsWithSorting, columnsByName]);
    var reducer = createReducer();
    var _a = useReducer(reducer, {
        columns: columnsWithSorting,
        columnsByName: columnsByName,
        originalRows: tableData,
        rows: tableData,
        selectedRows: [],
        toggleAllState: false,
        filterOn: !!(options === null || options === void 0 ? void 0 : options.filter),
        sortColumn: options === null || options === void 0 ? void 0 : options.sortColumn,
        paginationEnabled: !!(options === null || options === void 0 ? void 0 : options.pagination),
        pagination: {
            page: 1,
            perPage: 10,
            canNext: true,
            canPrev: false,
            // eslint-disable-next-line @typescript-eslint/no-empty-function
            nextPage: noop,
            // eslint-disable-next-line @typescript-eslint/no-empty-function
            prevPage: noop,
        },
    }), state = _a[0], dispatch = _a[1];
    state.pagination.nextPage = useCallback(function () {
        dispatch({ type: "NEXT_PAGE" });
    }, [dispatch]);
    state.pagination.prevPage = useCallback(function () { return dispatch({ type: "PREV_PAGE" }); }, [dispatch]);
    useEffect(function () {
        dispatch({ type: "SET_ROWS", data: tableData });
    }, [tableData]);
    var headers = useMemo(function () {
        return __spreadArrays(state.columns.map(function (column) {
            var label = column.label ? column.label : column.name;
            return __assign(__assign({}, column), { render: makeHeaderRender(label, column.headerRender) });
        }));
    }, [state.columns]);
    useEffect(function () {
        if (options && options.filter) {
            dispatch({ type: "GLOBAL_FILTER", filter: options.filter });
        }
    });
    return {
        headers: headers.filter(function (column) { return !column.hidden; }),
        rows: state.rows,
        originalRows: state.originalRows,
        selectedRows: state.selectedRows,
        dispatch: dispatch,
        selectRow: function (rowId) { return dispatch({ type: "SELECT_ROW", rowId: rowId }); },
        toggleAll: function () { return dispatch({ type: "TOGGLE_ALL" }); },
        toggleSort: function (columnName, isAscOverride) {
            return dispatch({ type: "TOGGLE_SORT", columnName: columnName, isAscOverride: isAscOverride });
        },
        setSearchString: function (searchString) { return dispatch({ type: "SEARCH_STRING", searchString: searchString }); },
        pagination: state.pagination,
        toggleAllState: state.toggleAllState,
    };
};

var StyledInput = styled(Input)(templateObject_1$e || (templateObject_1$e = __makeTemplateObject(["\n  border-radius: 16px;\n  margin-left: auto;\n"], ["\n  border-radius: 16px;\n  margin-left: auto;\n"])));
var InputWrapper = styled.div(templateObject_2$3 || (templateObject_2$3 = __makeTemplateObject(["\n  position: relative;\n  display: block;\n"], ["\n  position: relative;\n  display: block;\n"])));
var SearchInput = function (_a) {
    var onChangeCallback = _a.onChange, _b = _a.placeholder, placeholder = _b === void 0 ? "Search" : _b;
    var _c = useState(""), searchText = _c[0], setSearchText = _c[1];
    var debouncedOnChange = useMemo(function () { return debounce(function (e) { return onChangeCallback(e); }, 500); }, [onChangeCallback]);
    var onChange = function (e) {
        setSearchText(e.target.value);
        debouncedOnChange(e);
    };
    return (React.createElement(InputWrapper, null,
        React.createElement(StyledInput, { value: searchText, onChange: onChange, placeholder: placeholder })));
};
var templateObject_1$e, templateObject_2$3;

var DropDownHeader = styled.div(templateObject_1$f || (templateObject_1$f = __makeTemplateObject(["\n  width: 100%;\n  height: 40px;\n  display: flex;\n  align-items: center;\n  justify-content: space-between;\n  padding: 0px 16px;\n\n  border-radius: 16px;\n\n  transition: border-radius 0.15s;\n"], ["\n  width: 100%;\n  height: 40px;\n  display: flex;\n  align-items: center;\n  justify-content: space-between;\n  padding: 0px 16px;\n\n  border-radius: 16px;\n\n  transition: border-radius 0.15s;\n"])));
var DropDownListContainer = styled.div(templateObject_2$4 || (templateObject_2$4 = __makeTemplateObject(["\n  min-width: 150px;\n  height: 0;\n  position: absolute;\n  overflow: hidden;\n\n  transition: transform 0.15s, opacity 0.15s;\n  transform: scaleY(0);\n  transform-origin: top;\n  opacity: 0;\n  width: 100%;\n  /* background: #fffdf6; */\n  box-shadow: 0px 4px 10px rgba(208, 214, 227, 0.7);\n"], ["\n  min-width: 150px;\n  height: 0;\n  position: absolute;\n  overflow: hidden;\n\n  transition: transform 0.15s, opacity 0.15s;\n  transform: scaleY(0);\n  transform-origin: top;\n  opacity: 0;\n  width: 100%;\n  /* background: #fffdf6; */\n  box-shadow: 0px 4px 10px rgba(208, 214, 227, 0.7);\n"])));
var DropDownContainer = styled(Box)(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n  cursor: pointer;\n  width: 100%;\n  position: relative;\n\n  border-radius: 16px;\n  height: 40px;\n  min-width: 134px;\n  user-select: none;\n  z-index: 20;\n\n  ", "\n\n  svg {\n    position: absolute;\n    right: 16px;\n    top: 50%;\n    transform: translateY(-50%);\n  }\n"], ["\n  cursor: pointer;\n  width: 100%;\n  position: relative;\n\n  border-radius: 16px;\n  height: 40px;\n  min-width: 134px;\n  user-select: none;\n  z-index: 20;\n\n  ",
    "\n\n  svg {\n    position: absolute;\n    right: 16px;\n    top: 50%;\n    transform: translateY(-50%);\n  }\n"])), function (props) {
    return props.isOpen && css(templateObject_3$1 || (templateObject_3$1 = __makeTemplateObject(["\n      ", " {\n        border-radius: 16px 16px 0 0;\n      }\n\n      ", " {\n        height: auto;\n        transform: scaleY(1);\n        opacity: 1;\n\n        border-top-width: 0;\n        border-radius: 0 0 16px 16px;\n      }\n    "], ["\n      ", " {\n        border-radius: 16px 16px 0 0;\n      }\n\n      ", " {\n        height: auto;\n        transform: scaleY(1);\n        opacity: 1;\n\n        border-top-width: 0;\n        border-radius: 0 0 16px 16px;\n      }\n    "])), DropDownHeader, DropDownListContainer);
});
var DropDownList = styled.ul(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n  padding: 0;\n  margin: 0;\n  box-sizing: border-box;\n"], ["\n  padding: 0;\n  margin: 0;\n  box-sizing: border-box;\n"])));
var ListItem = styled.li(templateObject_6 || (templateObject_6 = __makeTemplateObject(["\n  list-style: none;\n  padding: 8px 16px;\n  transition: all 0.2s ease;\n  &:hover {\n    .text-select {\n      color: #ffc42a;\n    }\n  }\n"], ["\n  list-style: none;\n  padding: 8px 16px;\n  transition: all 0.2s ease;\n  &:hover {\n    .text-select {\n      color: #ffc42a;\n    }\n  }\n"])));
var Select = function (_a) {
    var options = _a.options, onOptionChange = _a.onOptionChange, _b = _a.defaultOptionIndex, defaultOptionIndex = _b === void 0 ? 0 : _b, props = __rest(_a, ["options", "onOptionChange", "defaultOptionIndex"]);
    var dropdownRef = useRef(null);
    var _c = useState(false), isOpen = _c[0], setIsOpen = _c[1];
    var _d = useState(defaultOptionIndex), selectedOptionIndex = _d[0], setSelectedOptionIndex = _d[1];
    var toggling = function (event) {
        setIsOpen(!isOpen);
        event.stopPropagation();
    };
    var onOptionClicked = function (selectedIndex) { return function () {
        setSelectedOptionIndex(selectedIndex);
        setIsOpen(false);
        if (onOptionChange) {
            onOptionChange(options[selectedIndex]);
        }
    }; };
    useEffect(function () {
        var handleClickOutside = function () {
            setIsOpen(false);
        };
        document.addEventListener("click", handleClickOutside);
        return function () {
            document.removeEventListener("click", handleClickOutside);
        };
    }, []);
    return (React.createElement(DropDownContainer, __assign({ isOpen: isOpen }, props),
        React.createElement(DropDownHeader, { onClick: toggling },
            React.createElement(Flex, null,
                React.createElement("img", { src: options[selectedOptionIndex].img, alt: "", style: { paddingRight: "4px" } }),
                React.createElement(Text, { bold: true }, options[selectedOptionIndex].label))),
        React.createElement(Icon$8, { fill: "#4F4F4F", onClick: toggling }),
        React.createElement(DropDownListContainer, { className: "cus-dropdown" },
            React.createElement(DropDownList, { ref: dropdownRef }, options.map(function (option, index) {
                return index !== selectedOptionIndex ? (React.createElement(ListItem, { onClick: onOptionClicked(index), key: option.label },
                    React.createElement(Flex, null,
                        React.createElement("img", { src: option.img, alt: "" }),
                        React.createElement(Text, { className: "text-select" }, option.label)))) : null;
            })))));
};
var templateObject_1$f, templateObject_2$4, templateObject_3$1, templateObject_4, templateObject_5, templateObject_6;

var breakpointMap = {
    xs: 390,
    sm: 576,
    mdsmall: 768,
    md: 820,
    lg: 968,
    xl: 1080,
    xxl: 1200,
};
var breakpoints = Object.values(breakpointMap).map(function (breakpoint) { return breakpoint + "px"; });
var mediaQueries = {
    xs: "@media screen and (max-width: " + breakpointMap.xs + "px)",
    sm: "@media screen and (max-width: " + breakpointMap.sm + "px)",
    md: "@media screen and (max-width: " + breakpointMap.md + "px)",
    mdSmall: "@media screen and (max-width: " + breakpointMap.mdSmall + "px)",
    lg: "@media screen and (max-width: " + breakpointMap.lg + "px)",
    xl: "@media screen and (max-width: " + breakpointMap.xl + "px)",
    xxl: "@media screen and (max-width: " + breakpointMap.xxl + "px)",
    nav: "@media screen and (max-width: " + breakpointMap.lg + "px)",
};
// const logger = {
//   breakpoints,
//   mediaQueries,
// };
// export default logger;
var base = {
    breakpoints: breakpoints,
    mediaQueries: mediaQueries,
};

var mediaQueries$1 = (function () {
    var prevMinWidth = 0;
    return Object.keys(breakpointMap).reduce(function (accum, size, index) {
        var _a, _b;
        if (index === Object.keys(breakpointMap).length - 1) {
            return __assign(__assign({}, accum), (_a = {}, _a[size] = "(max-width: " + prevMinWidth + "px)", _a));
        }
        // const minWidth = prevMinWidth;
        var breakpoint = breakpointMap[size];
        prevMinWidth = breakpoint + 1;
        return __assign(__assign({}, accum), (_b = {}, _b[size] = "(min-width: 0px) and (max-width: " + breakpoint + "px)", _b));
    }, {});
})();
var getKey = function (size) { return "is" + size.charAt(0).toUpperCase() + size.slice(1); };
var useMatchBreakpoints = function () {
    var _a = useState(function () {
        return Object.keys(mediaQueries$1).reduce(function (accum, size) {
            var _a;
            var key = getKey(size);
            var mql = window.matchMedia(mediaQueries$1[size]);
            return __assign(__assign({}, accum), (_a = {}, _a[key] = mql.matches, _a));
        }, {});
    }), state = _a[0], setState = _a[1];
    useEffect(function () {
        var handlers = Object.keys(mediaQueries$1).map(function (size) {
            var mql = window.matchMedia(mediaQueries$1[size]);
            var handler = function (matchMediaQuery) {
                var key = getKey(size);
                setState(function (prevState) {
                    var _a;
                    return (__assign(__assign({}, prevState), (_a = {}, _a[key] = matchMediaQuery.matches, _a)));
                });
            };
            if (mql.addEventListener) {
                mql.addEventListener("change", handler);
            }
            return function () {
                if (mql.removeEventListener) {
                    mql.removeEventListener("change", handler);
                }
            };
        });
        return function () {
            handlers.forEach(function (unsubscribe) {
                unsubscribe();
            });
        };
    }, [setState]);
    return __assign(__assign({}, state), { isMobile: state.isXs, isMobileL: state.isSm, isTabletS: state.isMdsmall, isTablet: state.isMd, isTabletL: state.isLg, isDesktop: state.isXl, isDesktopL: state.isXxl });
};
// 40 52 64

// import styled from "styled-components";
var Overlay = styled.div.attrs({ role: "presentation" })(templateObject_1$g || (templateObject_1$g = __makeTemplateObject(["\n  position: fixed;\n  top: 0px;\n  left: 0px;\n  width: 100%;\n  height: 100%;\n  background-color: rgba(0, 0, 0, 0.5);\n  /* transition: opacity 0.4s; */\n  opacity: ", ";\n  z-index: ", ";\n  pointer-events: ", ";\n"], ["\n  position: fixed;\n  top: 0px;\n  left: 0px;\n  width: 100%;\n  height: 100%;\n  background-color: rgba(0, 0, 0, 0.5);\n  /* transition: opacity 0.4s; */\n  opacity: ", ";\n  z-index: ", ";\n  pointer-events: ", ";\n"])), function (_a) {
    var show = _a.show;
    return (show ? 0.6 : 0);
}, function (_a) {
    var zIndex = _a.zIndex;
    return zIndex;
}, function (_a) {
    var show = _a.show;
    return (show ? "initial" : "none");
});
Overlay.defaultProps = {
    show: false,
    zIndex: 0,
};
var templateObject_1$g;

var opaci = keyframes(templateObject_1$h || (templateObject_1$h = __makeTemplateObject(["\n  0% {\n    opacity: 0;\n  }\n\n  100% {\n    opacity: 1;\n    \n  }\n"], ["\n  0% {\n    opacity: 0;\n  }\n\n  100% {\n    opacity: 1;\n    \n  }\n"])));
var ModalWrapper = styled.div(templateObject_2$5 || (templateObject_2$5 = __makeTemplateObject(["\n  display: flex;\n  flex-direction: column;\n  justify-content: center;\n  align-items: center;\n  position: fixed;\n  top: 0;\n  right: 0;\n  bottom: 0;\n  left: 0;\n  z-index: 99;\n  animation: 0.3s ease-in-out 0s 1 normal forwards running ", ";\n  will-change: opacity;\n  opacity: 0;\n  /* position: relative; */\n  /* width: 100%;\n  height: 100%; */\n"], ["\n  display: flex;\n  flex-direction: column;\n  justify-content: center;\n  align-items: center;\n  position: fixed;\n  top: 0;\n  right: 0;\n  bottom: 0;\n  left: 0;\n  z-index: 99;\n  animation: 0.3s ease-in-out 0s 1 normal forwards running ", ";\n  will-change: opacity;\n  opacity: 0;\n  /* position: relative; */\n  /* width: 100%;\n  height: 100%; */\n"])), opaci);
var Context = createContext({
    onPresent: function () { return null; },
    onDismiss: function () { return null; },
    setCloseOnOverlayClick: function () { return true; },
});
var ModalProvider = function (_a) {
    var children = _a.children;
    var _b = useState(false), isOpen = _b[0], setIsOpen = _b[1];
    var _c = useState(), modalNode = _c[0], setModalNode = _c[1];
    var _d = useState(true), closeOnOverlayClick = _d[0], setCloseOnOverlayClick = _d[1];
    var handlePresent = function (node) {
        setModalNode(node);
        setIsOpen(true);
    };
    var handleDismiss = function () {
        setModalNode(undefined);
        setIsOpen(false);
    };
    var handleOverlayDismiss = function () {
        if (closeOnOverlayClick) {
            handleDismiss();
        }
    };
    return (React.createElement(Context.Provider, { value: {
            onPresent: handlePresent,
            onDismiss: handleDismiss,
            setCloseOnOverlayClick: setCloseOnOverlayClick,
        } },
        isOpen && (React.createElement(ModalWrapper, null,
            React.createElement(Overlay, { show: true, onClick: handleOverlayDismiss }),
            React.isValidElement(modalNode) &&
                React.cloneElement(modalNode, {
                    onDismiss: handleDismiss,
                }))),
        children));
};
var templateObject_1$h, templateObject_2$5;

var useModal = function (modal, closeOnOverlayClick) {
    if (closeOnOverlayClick === void 0) { closeOnOverlayClick = true; }
    var _a = useContext(Context), onPresent = _a.onPresent, onDismiss = _a.onDismiss, setCloseOnOverlayClick = _a.setCloseOnOverlayClick;
    var onPresentCallback = useCallback(function () {
        onPresent(modal);
    }, [modal, onPresent]);
    useEffect(function () {
        setCloseOnOverlayClick(closeOnOverlayClick);
    }, [closeOnOverlayClick, setCloseOnOverlayClick]);
    return [onPresentCallback, onDismiss];
};

var ResetCSS = createGlobalStyle(templateObject_1$i || (templateObject_1$i = __makeTemplateObject(["\n  /* prettier-ignore */\n  html, body, div, span, applet, object, iframe,\n  h1, h2, h3, h4, h5, h6, p, blockquote, pre,\n  a, abbr, acronym, address, big, cite, code,\n  del, dfn, em, img, ins, kbd, q, s, samp,\n  small, strike, strong, sub, sup, tt, var,\n  b, u, i, center,\n  dl, dt, dd, ol, ul, li,\n  fieldset, form, label, legend,\n  table, caption, tbody, tfoot, thead, tr, th, td,\n  article, aside, canvas, details, embed, \n  figure, figcaption, footer, header, hgroup, \n  menu, nav, output, ruby, section, summary,\n  time, mark, audio, video {\n    margin: 0;\n    padding: 0;\n    border: 0;\n    font-size: 100%;\n    vertical-align: baseline;\n  }\n  /* HTML5 display-role reset for older browsers */\n  /* prettier-ignore */\n  article, aside, details, figcaption, figure, \n  footer, header, hgroup, menu, nav, section {\n    display: block;\n  }\n  body {\n    line-height: 1;\n    font-size: 16px;\n  }\n  ol,\n  ul {\n    list-style: disc;\n    list-style-position: inside;\n  }\n  blockquote,\n  q {\n    quotes: none;\n  }\n  blockquote:before,\n  blockquote:after,\n  q:before,\n  q:after {\n    content: \"\";\n    content: none;\n  }\n  table {\n    border-collapse: collapse;\n    border-spacing: 0;\n  }\n  a {\n    color: inherit;\n    text-decoration: none;\n  }\n  [role=\"button\"] {\n    cursor: pointer;\n  }\n  *,\n  *::before,\n  *::after {\n    box-sizing: border-box;\n  }\n  * {\n    font-family: 'Kanit', sans-serif;\n    -webkit-font-smoothing: antialiased;\n    -moz-osx-font-smoothing: grayscale;\n  }\n\n  /* Number */\n  input::-webkit-outer-spin-button,\n  input::-webkit-inner-spin-button {\n    -webkit-appearance: none;\n    margin: 0;\n  }\n  input[type=number] {\n    -moz-appearance: textfield;\n  }\n\n  /* Scrollbar */\n  ::-webkit-scrollbar {\n    width: 8px;\n  }\n  ::-webkit-scrollbar-thumb {\n    background: #F7DA57;; \n    border-radius: 8px;\n  }\n  ::-webkit-scrollbar-track {\n    box-shadow: #F7DA57;; \n    border-radius: 10px;\n  }\n\n  /* Slider */ \n  input[type=range] {\n    -webkit-appearance: none; /* Hides the slider so that custom slider can be made */\n    width: 100%; /* Specific width is required for Firefox. */\n    background: transparent; /* Otherwise white in Chrome */\n  }\n  input[type=range]::-webkit-slider-thumb {\n    -webkit-appearance: none;\n  }\n  input[type=range]:focus {\n    outline: none; /* Removes the blue border. You should probably do some kind of focus styling for accessibility reasons though. */\n  }\n  input[type=range]::-ms-track {\n    width: 100%;\n    cursor: pointer;\n    /* Hides the slider so custom styles can be added */\n    background: transparent; \n    border-color: transparent;\n    color: transparent;\n  }  \n"], ["\n  /* prettier-ignore */\n  html, body, div, span, applet, object, iframe,\n  h1, h2, h3, h4, h5, h6, p, blockquote, pre,\n  a, abbr, acronym, address, big, cite, code,\n  del, dfn, em, img, ins, kbd, q, s, samp,\n  small, strike, strong, sub, sup, tt, var,\n  b, u, i, center,\n  dl, dt, dd, ol, ul, li,\n  fieldset, form, label, legend,\n  table, caption, tbody, tfoot, thead, tr, th, td,\n  article, aside, canvas, details, embed, \n  figure, figcaption, footer, header, hgroup, \n  menu, nav, output, ruby, section, summary,\n  time, mark, audio, video {\n    margin: 0;\n    padding: 0;\n    border: 0;\n    font-size: 100%;\n    vertical-align: baseline;\n  }\n  /* HTML5 display-role reset for older browsers */\n  /* prettier-ignore */\n  article, aside, details, figcaption, figure, \n  footer, header, hgroup, menu, nav, section {\n    display: block;\n  }\n  body {\n    line-height: 1;\n    font-size: 16px;\n  }\n  ol,\n  ul {\n    list-style: disc;\n    list-style-position: inside;\n  }\n  blockquote,\n  q {\n    quotes: none;\n  }\n  blockquote:before,\n  blockquote:after,\n  q:before,\n  q:after {\n    content: \"\";\n    content: none;\n  }\n  table {\n    border-collapse: collapse;\n    border-spacing: 0;\n  }\n  a {\n    color: inherit;\n    text-decoration: none;\n  }\n  [role=\"button\"] {\n    cursor: pointer;\n  }\n  *,\n  *::before,\n  *::after {\n    box-sizing: border-box;\n  }\n  * {\n    font-family: 'Kanit', sans-serif;\n    -webkit-font-smoothing: antialiased;\n    -moz-osx-font-smoothing: grayscale;\n  }\n\n  /* Number */\n  input::-webkit-outer-spin-button,\n  input::-webkit-inner-spin-button {\n    -webkit-appearance: none;\n    margin: 0;\n  }\n  input[type=number] {\n    -moz-appearance: textfield;\n  }\n\n  /* Scrollbar */\n  ::-webkit-scrollbar {\n    width: 8px;\n  }\n  ::-webkit-scrollbar-thumb {\n    background: #F7DA57;; \n    border-radius: 8px;\n  }\n  ::-webkit-scrollbar-track {\n    box-shadow: #F7DA57;; \n    border-radius: 10px;\n  }\n\n  /* Slider */ \n  input[type=range] {\n    -webkit-appearance: none; /* Hides the slider so that custom slider can be made */\n    width: 100%; /* Specific width is required for Firefox. */\n    background: transparent; /* Otherwise white in Chrome */\n  }\n  input[type=range]::-webkit-slider-thumb {\n    -webkit-appearance: none;\n  }\n  input[type=range]:focus {\n    outline: none; /* Removes the blue border. You should probably do some kind of focus styling for accessibility reasons though. */\n  }\n  input[type=range]::-ms-track {\n    width: 100%;\n    cursor: pointer;\n    /* Hides the slider so custom styles can be added */\n    background: transparent; \n    border-color: transparent;\n    color: transparent;\n  }  \n"])));
var templateObject_1$i;

export { Icon$6 as AffiliatesIcon, Icon$8 as ArrowDropDownIcon, Box, ButtonMenu, ButtonMenuItem, Button as ButtonV1, Icon as ChevronDownIcon, Icon$1 as ChevronUpIcon, Icon$2 as DashboardIcon, Icon$4 as DepositIcon, Flex, Heading, IconButton, Input, InputGroup, Icon$3 as InvestmentIcon, Icon$9 as MaThongBaoIcon, ModalProvider, ResetCSS, SearchInput, Select, Icon$7 as SettingsIcon, Svg, Table, Tds, Text, Ths, TooltipText, Icon$5 as TransactionIcon, byTextAscending, byTextDescending, makeRender, base as mediaQueries, useMatchBreakpoints, useModal, useTable };
